awx/ui/src/components/ListHeader/ListHeader.js

```diff
@@ -56,6 +57,11 @@ function ListHeader(props) {
   };
 
   const handleRemove = (key, value) => {
+    if (key === 'ansible_facts') {
+      value = `${key}__${value}`;
+      key = 'host_filter';
+    }
+
```

Comment:
I think this logic would make more sense inside Search, since that's where we do the reverse work when we call parseQueryString(). I think the fewer components that need to be aware of this special logic, the better